### ISSUE TITLE
Xcode warning: [UIApplication registerForRemoteNotifications] must be used from main thread only

### ISSUE THREAD

**Issue by antailyaqwer** [mirzemehdi/KMPNotifier#37]
Hey, we have a warning inside working app (logs from XCode)
```
/Users/runner/work/KMPNotifier/KMPNotifier/kmpnotifier/src/iosMain/kotlin/com/mmk/kmpnotifier/firebase/FirebasePushNotifierImpl.kt:20 -[UIApplication registerForRemoteNotifications] must be used from main thread only
```

As I can see, this warning may lead to unexpected behavior, please, check it out

**Comment by mirzemehdi** [comment]
Thanks for letting me know @antailyaqwer

**Comment by coletz** [comment]
Sorry for botherin @mirzemehdi , any news on this? I'd only like to know if this will be solved in the next few days or if I need to fork the library to try to fix it on my own. Thanks!

**Comment by mirzemehdi** [comment]
@coletz  Planning to make a new release this week/weekend, though PR are always welcome :)

**Comment by coletz** [comment]
Ok, I'll try to dig into it. Can't understand why I'm getting this error tho,  from the stacktrace it looks like  it's caused by some lazy initialization done through koin

**Comment by antailyaqwer** [comment]
> Ok, I'll try to dig into it. Can't understand why I'm getting this error tho, from the stacktrace it looks like it's caused by some lazy initialization done through koin

Are you sure the problem is with this warning?
As it's a warning, it doesn't really prevents to launch the app.

**Comment by coletz** [comment]
> > Ok, I'll try to dig into it. Can't understand why I'm getting this error tho, from the stacktrace it looks like it's caused by some lazy initialization done through koin
> 
> Are you sure the problem is with this warning? As it's a warning, it doesn't really prevents to launch the app.

Sorry I meant warning, not error. It is not preventing the app to start, but it's not registering correctly to fcm and the token is not synced with apns (at least that's what I'm seeing). I won't be able to work on this before Saturday probably

**Comment by mirzemehdi** [comment]
@antailyaqwer where you call initialize method of the library? Because I can't reproduce this warning

**Comment by mirzemehdi** [comment]
Should be fixed in version `1.0.1`
